pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -265,7 +265,7 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 				if portProtocols[port.Port] != "" && portProtocols[port.Port] != protocol.TCP {
 					portProtocols[port.Port] = protocol.Unsupported
 				} else {
-					portProtocols[port.Port] = protocol.TCP
+					portProtocols[port.Port] = port.Protocol
```

Comment:
I think this will cause issues with the L265 above, where its checking only for TCP?